server/game/cards/01_SOR/units/EzraBridger.ts

```diff
@@ -0,0 +1,42 @@
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { PlayType, TargetMode } from '../../../core/Constants';
+import AbilityHelper from '../../../AbilityHelper';
+
+
+export default class EzraBridger extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '9560139036',
+            internalName: 'ezra-bridger#resourceful-troublemaker'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addTriggeredAbility({
+            title: 'Look at the top card of your deck.',
+            when: {
+                onAttackCompleted: (event, context) => event.attack.attacker === context.source,
+            },
+            immediateEffect: AbilityHelper.immediateEffects.lookAt(
+                (context) => ({ target: context.source.controller.getTopCardOfDeck() })
+            ),
+            then: (thenContext) => {
+                const topCardOfDeck = thenContext.source.controller.getTopCardOfDeck();
+                return {
+                    title: 'You may play it, discard it, or leave it on top of your deck.',
+                    thenCondition: (context) => context.source.controller.drawDeck.length > 0,
```

Comment:
I've had a realization recently that cards of the form "look at the top card of your deck. do [X] with it" might actually be best represented with `ifYouDo` instead of `then` with a `thenCondition`. The card doesn't explicitly say "if you do" but mechanically it's understood that if you don't look at the top card for any reason (including the deck being empty) then the second part of the effect should fizzle.

Could you try making that change here? There is an option to have an `ifYouDoContext` but I think you won't need it since you're just reading context.source. If you do want to use it, you might need to merge main since I just did a bug fix for it